### PR TITLE
[PortfolioPerformance] Ensure that parsed activities are valid through `validateActivity`. Adjust function `csvJSON` to remove leading and the trailing `"`

### LINKED ISSUE #107: Portfolio Performance CSV Bug at WKN
In some occassions, the WKN field is stored like this `"884437 "` which will return no results from the Tresor One search. It should be stored like this: `884437`.

Example data
```
Datum;Typ;Wert;Buchungswährung;Bruttobetrag;Währung Bruttobetrag;Wechselkurs;Gebühren;Steuern;Stück;ISIN;WKN;Ticker-Symbol;Wertpapiername;Notiz
2017-08-07T00:00;Kauf;2.040,09;EUR;;;;10,49;0,00;43;;884437;;Starbucks;
2020-02-28T00:00;Verkauf;-2.620,46;EUR;;;;14,59;241,95;42;;884437;;Starbucks;
2020-08-21T00:00;Kauf;5.031,13;EUR;;;;19,93;0,00;48;;865985;;Apple;
```

### PR DESCRIPTION
**Pull request by dredav**
- This will fix #107 

**global function `validateActivity`**
- Function ensure that ISIN or WKN exists
- Function validate that the WKN is valid
- Add additional test cases

**global function `csvJSON`**
- Leading and the trailing `"` will be removed from csv entry

**parser for portfolio performance**
- All activities will now be validated through global function `validateActivity`
- When the fields `isin`, `wkn`, `symbol` or `grossCurrency` is `undefined` or has an empty value, the fields will removed from activity (to be valid for global function `validateActivity`)
- Remove `console.log` entries
- Add additional test cases

### CODE REVIEW

File: src/apps/portfolioPerformance/index.js [hunk 1]
```diff
@@ -23,17 +23,30 @@ const activityNormalizer = typeKeyMap => activity => {
     {}
   );
```
Comment (codeofsumit) [inline src/apps/portfolioPerformance/index.js:24]: Can we add an explanatory comment here on what this is doing overall and why?

Comment (codeofsumit) [inline src/apps/portfolioPerformance/index.js:24]: My previous comment should reference to line 33

**Review by codeofsumit — COMMENTED** [review]
Just a test review, the github app seems to have issues

**Comment by codeofsumit** [comment]
My github app is throwing errors when I try to do a review. Will do a review asap.

**Comment by codeofsumit** [comment]
Amazing work man!

Just a few minor things.

Have to comment here because GitHub line comments don't work:

helper/index Line 117:

The Tresor One search can handle searches by name as well. Therefore, WKN and ISIN is not required when extracting activities from PP. The Import (in T1) will automatically use the `name` field and tries to find a security for it.
If non (or more than 1) are found, it will prompt the user to search and assign a security.

=> change to `must have at least ISIN, WKN or Name`



helper/index line 19
Please add an explanatory comment above the replace.trim what exactly is replaced and trimmed

**Comment by dredav** [comment]
@codeofsumit Thanks for your review! 

>The Tresor One search can handle searches by name as well. Therefore, WKN and ISIN is not required when extracting activities from PP

Does this apply to all brokers or only to PP?

**Comment by codeofsumit** [comment]
Yes, `company` - sorry.

The search in T1 is independent from the import so yes, theoretically this applies to all PDFs.
The difference is that we currently expect PDFs to have an ISIN or WKN, which is reasonable. So no need to change this for all brokers IMO (but it's possible from a T1 perspective).
For PP it's important because users manually created these entries while PDFs are generated from brokers (so more reliable data in them).

**Comment by dredav** [comment]
@codeofsumit done. The parser for PP will now check for `.company`. All other parser will "ignore" (only check if the field is not empty when it exists) this field at the `validateActivity` check.

**Comment by codeofsumit** [comment]
awesome! With these changes, the provided CSV should work fine, right?

**Comment by dredav** [comment]
Yes and no. I can't reproduce the bug with your .csv from #107. So I've added a test case with a different csv which I generated from PP with a space at the end of the WKN. This test case can now be processed with this PR.

**Pull request merged**